### PR TITLE
Systemd code cleanup

### PR DESCRIPTION
**Pull request by kgaillot**
Most significantly, this handles DBus errors better. It's also intended to make future work on the systemd code easier.

### CODE REVIEW

**Comment by wenningerk** [comment]
https://dbus.freedesktop.org/doc/api/html/group__DBusConnection.html#ga55ff88cd22c0672441c7deffbfb68fbf
is quite direct in that  dbus_connection_dispatch(connection) shouldn't be called from within the 
function registered.
So I guess we should address this here as well ...

File: lib/services/systemd.c [hunk 1]
```diff
@@ -701,10 +698,7 @@ systemd_unit_exec_with_unit(svc_action_t * op, const char *unit)
         }
 
     } else {
-        DBusError error;
-
-        reply = systemd_send_recv(msg, &error, op->timeout);
-        dbus_error_free(&error);
+        reply = systemd_send_recv(msg, NULL, op->timeout);
```
Comment (wenningerk) [inline lib/services/systemd.c]: with the error logging removed from pcmk_dbus_find_error wouldn't this lead to no error logged at all for this case?

**Comment by wenningerk** [comment]
There are still a few leftover dbus_error_init in systemd.c that could be removed for consistency as dbus.c should initialize the error on all calls.

**Comment by kgaillot** [comment]
> https://dbus.freedesktop.org/doc/api/html/group__DBusConnection.html#ga55ff88cd22c0672441c7deffbfb68fbf
> is quite direct in that dbus_connection_dispatch(connection) shouldn't be called from within the
> function registered.
> So I guess we should address this here as well ...

Good point. I think I'll do that in a separate request, as it will overlap with my signal changes.

**Comment by kgaillot** [comment]
> There are still a few leftover dbus_error_init in systemd.c that could be removed for consistency as dbus.c should initialize the error on all calls.

I thought about that, but I decided to leave them b/c there's no harm in initializing twice, and it would be a failsafe in case some future change to dbus.c misses initialization in some branch. But it is overkill, and it might be better to leave it to dbus.c, as long as we document it strongly there that all the functions must init their errors.

File: lib/services/pcmk-dbus.h [hunk 2]
```diff
@@ -15,5 +25,7 @@ char *pcmk_dbus_get_property(
     void (*callback)(const char *name, const char *value, void *userdata), void *userdata,
     DBusPendingCall **pending, int timeout);
 
-bool pcmk_dbus_find_error(const char *method, DBusPendingCall* pending, DBusMessage *reply, DBusError *error);
+bool pcmk_dbus_find_error(DBusPendingCall *pending, DBusMessage *reply,
+                          DBusError *error);
 
+#endif /* PCMK_DBUS__H */
```
Comment (kgaillot) [inline lib/services/pcmk-dbus.h]: Wouldn't hurt, but probably wouldn't be worth the time either.

**Pull request merged**